src/tools/r2rdump/R2RSignature.cs

```diff
@@ -460,7 +468,23 @@ public string ReadTypeSignature()
         /// <param name="builder"></param>
         private void ParseSignature(StringBuilder builder)
         {
-            uint fixupType = ReadUInt();
+            uint fixupType = PeekByte();
+            bool moduleOverride = (fixupType & (uint)CORCOMPILE_FIXUP_BLOB_KIND.ENCODE_MODULE_OVERRIDE) != 0;
+            // Check first byte for a module override being encoded
+            if (moduleOverride)
+            {
+                builder.Append("ENCODE_MODULE_OVERRIDE @ ");
+                fixupType &= ~(uint)CORCOMPILE_FIXUP_BLOB_KIND.ENCODE_MODULE_OVERRIDE;
+                ReadByte();
+                uint moduleIndex = ReadByte();
```

Comment:
I believe that module index should be read using ReadUInt corresponding to CoreCLR CorSigUncompressData, please see e.g. here:

https://github.com/dotnet/coreclr/blob/04c4df8c9ecf53499838945197af70f11ddf840a/src/vm/jitinterface.cpp#L13362

or

https://github.com/dotnet/coreclr/blob/04c4df8c9ecf53499838945197af70f11ddf840a/src/vm/genericdict.cpp#L673